crates/data/src/helpers/name_vec_alox.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2024 Melody Madeline Lyons
+//
+// This file is part of Luminol.
+//
+// Luminol is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Luminol is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Luminol.  If not, see <http://www.gnu.org/licenses/>.
+
+pub fn deserialize_with<'de, D>(deserializer: D) -> Result<[Option<String>; 7], alox_48::DeError>
```

Comment:
At this point, I feel like we should make a newtype over `Option<T>` that handles serialization instead of making more modules to handle it